test/Configuration.spec.js

```diff
@@ -16,8 +21,55 @@ describe("Configuration", function() {
           "Tag: Internal": ":house: Internal"
         },
         "cacheDir": ".changelog",
-        "rootPath": path.resolve(`${__dirname}/..`),
+        rootPath,
+      });
+    });
+  });
+
+  describe("fromPath", function() {
+    const tmpDir = `${os.tmpDir()}/changelog-test`;
+
+    beforeEach(function() {
+      fs.ensureDirSync(tmpDir);
+    });
+
+    afterEach(function() {
+      fs.removeSync(tmpDir);
+    });
+
+    it("reads the configuration from 'lerna.json'", function() {
+      fs.writeJsonSync(path.join(tmpDir, "lerna.json"), {
+        changelog: { repo: "foo/bar" },
+      });
+
+      const result = fromPath(tmpDir);
+      expect(result.repo).toEqual("foo/bar");
+    });
+
+    it("reads the configuration from 'package.json'", function() {
+      fs.writeJsonSync(path.join(tmpDir, "package.json"), {
+        changelog: { repo: "foo/bar" },
+      });
```

Comment:
Nitpick: you can nest describes to make it more readable, then put the setup in `beforeEach`. The `it` blocks ideally should only contain the expectations.

```js
describe('fromPath', () => {
  describe('when lerna.json is present', () => {
    beforeEach(() => {
      fs.writeJsonSync(path.join(tmpDir, 'lerna.json'), {
        changelog: { repo: 'foo/bar' },
      })
    })
    describe('the changelog configuration is loaded', () => {
      let result
      beforeEach(() => {
        result = fromPath(tmpDir)
      })
      it('has correct repo value', () => {
        expect(result.repo).toEqual('foo/bar')
      })
    })
  })
  describe('when package.json is present', () => {
    // same as before
  })
})
```